Adding OpenAlex works pipeline

Jack-Vines opened the pull request:
---
For completeness (as nesta_daps where this flow lives is still in a very beta phase), this PR adds the works pipeline I've used to generate OpenAlex API data for the project.

The flow is configurable and takes a list of concepts as well as years and produces a file for each year of search results from the API. It includes a production setting that defaults to False, in which case only the first concept and year in each list are used (for testing purposes).

Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [ ] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [x] Appropriate information has been added to `README`s
- [x] I have explained the feature in this PR or (better) in `output/reports/`
- [x] I have requested a code review

Jack-Vines committed:
Adding OpenAlex works pipeline

Jack-0-0 commented on ai_genomics/pipeline/openalex/works_pipeline.py:
```diff
@@ -0,0 +1,147 @@
+    query_list = []
+    for year in years:
+        query_list.append(f'{concepts_joined},publication_year:{year}')
+    return query_list
```
This can be simplified to:

```suggestion
return [f'{concepts_joined},publication_year:{year}' for year in years]
```

Jack-0-0 commented on ai_genomics/pipeline/openalex/works_pipeline.py:
```diff
@@ -0,0 +1,147 @@
+    all_pages = [
+        f"{API_ROOT}concepts.id:{concepts_text}&per-page=200&cursor="
+        for i in range(1, number_of_pages + 1)
+    ]
```
```suggestion
    all_pages = [
        f"{api_root}concepts.id:{concepts_text}&per-page=200&cursor="
        for _ in range(1, number_of_pages + 1)
    ]
```

Jack-0-0 reviewed (commented):
Thanks @Jack-Vines.

The code runs as expected and saves outputs to S3.

I have added some suggestions on specific lines.

Does this need a readme?

Jack-Vines committed:
Update to improve functions

Jack-Vines commented:
> Thanks @Jack-Vines.
> 
> The code runs as expected and saves outputs to S3.
> 
> I have added some suggestions on specific lines.
> 
> Does this need a readme?

Thanks @Jack-0-0, i've updated following suggestions. Regarding a README, i've tried to be pretty descriptive in the code, is there anything you think would be particularly useful from a README beyond what exists in docstrings/comments and the report?

Jack-0-0 reviewed (approved):
It was all clear to me. I was checking in case you wanted to add one 🙂
